### DISCUSSION TITLE
Send File. Help.

### DISCUSSION THREAD

**Discussion by PedroAlbuquerqueQueiroz** [viniciussanchez/RESTRequest4Delphi#116] [Q&A]
Vinicius, tudo bem ?

Por favor, com vejo que você tem um vasto conhecimento com restrequest, poderia me ajudar com a situação abaixo ?

-> Preciso enviar um arquivo em uma requisição post, porem o arquivo deve seguir no body em form-data:
file

foto: [https://www.jasaquei.com.br/docs/print_postman.png](url)

Tentativas

_Não deu certo.._
**LResponse := TRequest.New.BaseURL('Url')
                                     .AddHeader('Authorization','Bearer '+xTokenIntegracao,[poDoNotEncode])
                                     .AddParam('dataPagamento',formatdatetime('YYYY-MM-DD',Now),pkREQUESTBODY)
                                     .AddParam('codigoAutenticacao',DataSource1.DataSet.FieldByName('CWB_AUTENTICACAO_PAGTO').AsString,pkREQUESTBO
![print_postman](https://user-images.githubusercontent.com/53458137/202562392-a6c7955b-6cac-4cea-af53-cb39b4c783fe.png)
![Screenshot 2022-11-17 180358](https://user-images.githubusercontent.com/53458137/202562421-ec5dbe9e-d328-404d-bba6-e63f8ae1d67d.png)
DY)
                                     .AddParam('observacoes','Pagto Liquidado. '+formatdatetime('DD/MM/YY hh:mm',Now),pkREQUESTBODY)
                                     .AddFile(DataSource1.DataSet.FieldByName('CWB_ID').AsString+'.pdf',xStream)
                                     .Accept('application/json')
                                     .Post;**

_Não Deu certo (2)_
                    LRequest              := THTTPClient.Create;
//                  LFormData             := TMultipartFormData.Create();
//                  LResponseStringStream := TStringStream.Create;
//                  try
//                    LFormData.AddField('Authorization','Bearer '+xTokenIntegracao);
//                    LFormData.AddField('body',xBody);
//                    LFormData.AddFile('arquivo','C:\comprovantes\'+DataSource1.DataSet.FieldByName('CWB_ID').AsString+'.pdf'); 
//                    LRequest.Post('Url', LFormData, LResponseStringStream);
//                  finally
//                    LFormData.Free;
//                    LResponseStringStream.Free;
//                    LRequest.Free;
//                  end;

_Não Deu certo (3)_
RESTClient1.BaseURL := 'URL/'+FDMemTblIdLicenca.FieldByName('ID_LICENCIAMENTO').AsString;
                  RESTClient1.ContentType :=  'multipart/form-data';
                  RESTClient1.Accept :=  'multipart/form-data';

                  RESTRequest1.Params.AddHeader('Authorization','Bearer '+xTokenIntegracao);
                  RESTRequest1.Params.ParameterByName('Authorization').Options := [poDoNotEncode];

                  RESTRequest1.Params.AddItem;
                    RESTRequest1.Params.Items[1].Name := 'dataPagamento';
                    RESTRequest1.Params.Items[1].value := Copy(DataSource1.DataSet.FieldByName('CWB_DATAHORA_REGISTRO').AsString,1,10);
                    RESTRequest1.Params.Items[1].ContentType := ctMULTIPART_FORM_DATA;
                    RESTRequest1.Params.Items[1].Kind := TRESTRequestParameterKind.pkREQUESTBODY;

                  RESTRequest1.Params.AddItem;
                    RESTRequest1.Params.Items[2].Name := 'observacoes';
                    RESTRequest1.Params.Items[2].value :='Pagto Liquidado. '+Copy(DataSource1.DataSet.FieldByName('CWB_DATAHORA_REGISTRO').AsString,1,10);
                    RESTRequest1.Params.Items[2].ContentType := ctMULTIPART_FORM_DATA;
                    RESTRequest1.Params.Items[2].Kind := TRESTRequestParameterKind.pkREQUESTBODY;

                   RESTRequest1.Params.AddItem;
                    RESTRequest1.Params.Items[3].Name  := 'codigoAutenticacao';
                    RESTRequest1.Params.Items[3].value := DataSource1.DataSet.FieldByName('CWB_AUTENTICACAO_PAGTO').AsString;
                    RESTRequest1.Params.Items[3].ContentType := ctMULTIPART_FORM_DATA;
                    RESTRequest1.Params.Items[3].Kind := TRESTRequestParameterKind.pkREQUESTBODY;
                   RESTRequest1.AddFile('File','C:\comprovantes\'+DataSource1.DataSet.FieldByName('CWB_ID').AsString+'.pdf',DefaultRESTContentType.ctAPPLICATION_PDF);

                  RESTRequest1.Method := TRESTRequestMethod.rmPOST;
                  RESTRequest1.Execute;
                  RESTResponse1.StatusCode;
                  ShowMessage(RESTResponse1.Content);

Muito agradecido se poder ajudar.

**Comment by viniciussanchez** [comment]
Bom dia, se olhar no exemplo tem:

![image](https://user-images.githubusercontent.com/16382981/202693486-2b41b404-80cd-4ba4-b180-70490dc30c7c.png)

Vale dar uma olhada também nesse pull request:

https://github.com/viniciussanchez/RESTRequest4Delphi/pull/101
